File: src/2000.reverse-prefix-of-word.2128/solution.ts
function reversePrefix (word: string, ch: string): string {
    const index = word.indexOf(ch);
    if (index === -1) {
        return word;
    }
    return word.slice(0, index + 1).split('').reverse().join('') + word.slice(index + 1);
}

File: src/2000.reverse-prefix-of-word.2128/solution.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int r = word.indexOf(ch);
        int l = 0;
        char[] chars = word.toCharArray();
        while (l<r){
            char tmp = chars[l];
            chars[l++] = chars[r];
            chars[r--] = tmp;
        }
        return String.valueOf(chars);
    }
}